src/as3hx/Writer.hx

```diff
@@ -1276,7 +1276,14 @@ class Writer
                         f(def.el[def.el.length - 1], def.el);
                     }
                 }
-                newCases = loopCases(cases.slice(0), def == null ? null : def.el.slice(0), testVar, newCases);
+
+                if (def != null && def.before == null) {
+                    // default is in the end
+                    newCases = loopCases(cases.slice(0), def.el.copy(), testVar, newCases);
+                } else {
+                    // default is not in the end, so don't catch fall-through
+                    newCases = loopCases(cases.slice(0), null, testVar, newCases);
```

Comment:
`cases.slice(0)` -> `cases.copy()` ?

Reply:
So just you're aware I'm not sure what I'm changing here. Can you propose a test-case?